pkg/services/env_service.go

```diff
@@ -48,13 +91,84 @@ func ReadProxyProps(e *Env) error {
 		}
 
 		// Proxy CA certificate
-		caCertPath, caCertName, caCertData, err := prompts.ReadCACert("Proxy CA certificate filepath", e.ProxyCaCertPath, "")
+		if e.ProxyCACert == nil {
+			e.ProxyCACert = &CACert{}
+		}
+		caCertPath, caCertName, caCertData, err := prompts.ReadCACert("Proxy CA certificate filepath", e.ProxyCACert.Path, "")
 		if err != nil {
 			return err
 		}
-		e.ProxyCaCertData = string(caCertData)
-		e.ProxyCaCertName = caCertName
-		e.ProxyCaCertPath = caCertPath
+		e.ProxyCACert.Data = string(caCertData)
+		e.ProxyCACert.Name = caCertName
+		e.ProxyCACert.Path = caCertPath
+	}
+
+	return nil
+}
+
+// ReadHaulerProps prompts the user to configure hauler settings.
+func ReadHaulerProps(h *Hauler, e *Env) error {
+	var err error
+
+	// registry
+	if h.Host == "" {
+		h.Host = network.GetDefaultHostAddress()
+	}
+	h.Host, err = prompts.ReadText("Hauler Host (IPv4 address of primary NIC)", h.Host, false, -1)
+	if err != nil {
+		return err
+	}
+	if h.Port == 0 {
+		h.Port = 5000
+	}
+	h.Port, err = prompts.ReadInt("Hauler Port", fmt.Sprintf("%d", h.Port), 1024, 65535)
+	if err != nil {
+		return err
+	}
+
+	// basic auth
+	if h.BasicAuth == nil {
+		h.BasicAuth = &BasicAuth{}
+	}
+	h.BasicAuth.Username, h.BasicAuth.Password, err = prompts.ReadBasicCreds(
+		"Username", "Password", h.BasicAuth.Username, h.BasicAuth.Password, true, false,
+	)
+	if err != nil {
+		return err
+	}
+
+	// tls verification
+	h.InsecureSkipTLSVerify, err = prompts.ReadBool("Allow Insecure Connection (Bypass x509 Verification)", true)
+	if err != nil {
+		return err
+	}
+	if h.InsecureSkipTLSVerify {
+		return nil
+	}
+
+	// ca cert
+	if e.ProxyCACert.Path != "" {
+		h.ReuseProxyCACert, err = prompts.ReadBool("Reuse proxy CA cert for Hauler registry", true)
+		if err != nil {
+			return err
+		}
+	}
+	if h.CACert == nil {
+		h.CACert = &CACert{}
+	}
+	if h.ReuseProxyCACert {
+		h.CACert = e.ProxyCACert
+		return nil
+	}
+	caCertPath, caCertName, caCertData, err := prompts.ReadCACert("Hauler CA certificate filepath", h.CACert.Path, "")
+	if err != nil {
+		return err
+	} else if caCertPath == "" {
+		h = nil
+	} else {
+		h.CACert.Data = string(caCertData)
+		h.CACert.Name = caCertName
+		h.CACert.Path = caCertPath
 	}
```

Comment:
this is a total nit given how simple this code is, but imo this type of `if block` isnt ideal since it handles both error checking and business logic.

Setting it up like it is in the suggestion would be cleaner since it completley segregates the err handling and the business logic.

```suggestion
	if err != nil {
		return err
	}
	
	if caCertPath == "" {
		h = nil
	} else {
		h.CACert.Data = string(caCertData)
		h.CACert.Name = caCertName
		h.CACert.Path = caCertPath
	}
```